oidc_cli/storage/storage.go

```diff
@@ -4,11 +4,14 @@ import (
 	"context"
 
 	"github.com/chanzuckerberg/go-misc/oidc_cli/client"
+	"github.com/mitchellh/go-homedir"
+	"github.com/chanzuckerberg/go-misc/osutil"
+	"github.com/pkg/errors"
 )
 
 const (
 	service = "aws-oidc"
-
+	defaultFileStorageDir = "~/.oidc-cli"
```

Comment:
I'd suggest using the [XDG configuration directories](https://wiki.archlinux.org/title/XDG_Base_Directory) instead—they tend to be more portable and cause less clutter than hidden directories in the home dir. It's also worth pointing out that there are contexts where a user may not have a home directory (system users, service accounts, etc).

Following the XDG spec:

- User specific data goes in `$XDG_DATA_HOME`, which defaults to `$HOME/.local/share`
- User specific configuration data goes in `$XDG_CONFIG_HOME`, which defaults to `$HOME/.config`
- User specific cache files go in `$XDG_CACHE_HOME`, which defaults to `$HOME/.cache`

For an application called `oidc-cli`, your relevant application data would live in:

- `$XDG_DATA_HOME/oidc-cli`, which would usually be `~/.local/share/oidc-cli/`
- `$XDG_CONFIG_HOME/oidc-cli`, which would usually be `~/.config/oidc-cli`
- `$XDG_CACHE_HOME/oidc-cli`, which would usually be `~/.cache/oidc-coli`

You can use a library like https://github.com/adrg/xdg to figure out the appropriate application dirs if you don't want to fuss making your own implementation to find/create the relevant directories.

Reply:
I think that here, since this is aws-oidc specific code, we know the user has a home dir. I've changed the default path from ~/.oidc-cli to ~/.cache/oidc-cli to better represent what were using the storage for, and avoid creating clutter.

Reply:
:+1: , there are some built-ins stdlib as well I think -- https://pkg.go.dev/os#UserCacheDir